test/evp_test.c

```diff
@@ -1268,9 +1269,12 @@ static int mac_test_parse(EVP_TEST *t,
         return mdata->xof = 1;
     if (strcmp(keyword, "NoReinit") == 0)
         return mdata->no_reinit = 1;
-    if (strcmp(keyword, "Ctrl") == 0)
-        return sk_OPENSSL_STRING_push(mdata->controls,
-                                      OPENSSL_strdup(value)) != 0;
+    if (strcmp(keyword, "Ctrl") == 0) {
+        data = OPENSSL_strdup(value);
+        if (!data)
```

Comment:
The condition should be `data == NULL` instead.

Reply:
You can fix the `!mdata->alg` above too.

Reply:
Thanks, I have submitted a fixup to fix the above problems.